### PR TITLE
Fix #262 -- autolaunch the simulator.

### LINKED ISSUE #262: ios_sim.py or sky_tool or shelldb should launch the simulator for you
We shouldn't need a separate step in our instructions to launch the simulator.

We can run the "open" command from python instead of forcing the user to.

### PR DESCRIPTION
**Pull request by iansf**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
I wish we used `xcrun simctl boot` instead of looking for the binary directly. Seems more resilient to weird Xcode installs

File: build/config/ios/ios_sim.py [hunk 1]
```diff
@@ -4,11 +4,16 @@
...
 import errno
+import os
+import re
 import subprocess
 import sys
-import re
+import time
+
+IOS_SIM_PATH = [
+  '/Applications/iOS Simulator.app/Contents/MacOS/iOS Simulator'
```
Comment (eseidelGoogle) [inline build/config/ios/ios_sim.py]: I wonder if we even need this?  "open -a iOS\ Siumlator" should find it, etc.

File: build/config/ios/ios_sim.py [hunk 2]
```diff
@@ -32,8 +37,30 @@ def ApplicationIdentifier(path):
...
+  if IsDeviceBooted():
+    return
+
+  # Use Popen here because launching the simulator from the command line in this manner doesn't return, so we can't check the result.
+  if args.ios_sim_path:
+    subprocess.Popen( args.ios_sim_path )
+  else:
+    subprocess.Popen( IOS_SIM_PATH )
+
+  while not IsDeviceBooted():
```
Comment (eseidelGoogle) [inline build/config/ios/ios_sim.py]: Do you want a timeout?

**Comment by chinmaygarde** [comment]
You would also not have to wait an arbitrary amount of time for the device to boot

File: build/config/ios/ios_sim.py [hunk 3]
```diff
@@ -100,8 +127,15 @@ def Main():
                       default='localhost:8080',
                       help='Sky server address.')
 
+  parser.add_argument('--ios_sim_path', dest='ios_sim_path', required=False,
```
Comment (eseidelGoogle) [inline build/config/ios/ios_sim.py]: required is already false for all arguments starting with --.  You probably want this to be action='store_true'

File: build/config/ios/ios_sim.py [hunk 4]
```diff
@@ -100,8 +127,15 @@ def Main():
...
+                      help='Path to your iOS Simulator executable. '
+                      'Not normally required.')
+
   subparsers = parser.add_subparsers()
 
+  launch_parser = subparsers.add_parser('boot', help='Boot iOS Simulator')
```
Comment (eseidelGoogle) [inline build/config/ios/ios_sim.py]: I figured you would just boot implicitly as part of launch or debug?

**Comment by eseidelGoogle** [comment]
This looks good, but probably coudl go another round.  regardless lgtm.

File: build/config/ios/ios_sim.py (hunk 1 above)
Comment (iansf) [inline build/config/ios/ios_sim.py]: `open -a iOS\ Simulator` and similar variants don't work for me.  Does it work for you?

File: build/config/ios/ios_sim.py (hunk 4 above)
Comment (iansf) [inline build/config/ios/ios_sim.py]: I was testing with that.  I agree it's probably not needed as an explicit command.

**Comment by iansf** [comment]
@chinmaygarde: I was looking through the simctl docs and it wasn't immediately obvious to me that there was a set of simctl commands that would reliably give the user something sensible as Apple releases new simulators.  Did you have a particular plan in mind?  (`xcrun simctl boot` requires you to specify a particular simulator.)

Also, launching a particular simulator causes that simulator to be launched again the next time by default, so if we choose a preferred simulator, that may violate the principle of least surprise.  I agree though that if their default simulator is not one that Sky is compatible with (e.g., they have been testing very old versions of iOS), then the experience may be confusing.  Possibly we could launch the default simulator and then verify that it didn't launch a simulator that is incompatible with Sky?

**Pull request merged**